src/main/java/com/github/dockerjava/core/command/CreateContainerCmdImpl.java

```diff
@@ -125,11 +139,50 @@ public CreateContainerCmdImpl(CreateContainerCmd.Exec exec, String image) {
      */
     @Override
     public CreateContainerResponse exec() throws NotFoundException, ConflictException {
+        //code flow taken from https://github.com/docker/docker/blob/master/runconfig/opts/parse.go
```

Comment:
@fbuecklers I really understand your objection but please note that docker-java is just a library to drive the docker remote API. It shouldn't include any business logic that works on that API.

Emulating the docker CLI would possible on top of docker-java of course but nobody has started this yet. So the best we can do about is to document your findings in a testcase currently. Would that be OK for you?